src/hydration/hydration.ts

```diff
@@ -0,0 +1,127 @@
+import { DEFAULT_STALE_TIME, DEFAULT_CACHE_TIME } from '../core/config'
+
+import type { Query, QueryCache, QueryKey, QueryConfig } from 'react-query'
+
+export interface DehydratedQueryConfig<TResult> {
+  staleTime?: number
+  cacheTime?: number
+  initialData?: TResult
+}
+
+export interface DehydratedQuery<TResult> {
```

Comment:
Think the `<TResult>` is not needed anywhere in the hydration types as we don't really care about the actual data within the queries.

Reply:
I like that, it simplifies things. So what would that make the data? Unknown?

Reply:
I would say `unknown` if possible and otherwise just `any`.

Reply:
Fixed.